### ISSUE TITLE
False Positive: UnusedLocalVariableCheck

### ISSUE THREAD

**Issue by nburles** [checkstyle/checkstyle#11920]
In addition to the false positive described in #11374, I have found another one...

I have read check documentation: https://checkstyle.org/config_coding.html#UnusedLocalVariable
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedLocalVariable"/>
    </module>
</module>

/var/tmp $ cat Test.java
public class Test {
    void method() {
        final OuterClass outerClass = new OuterClass();
        final OuterClass.InnerClass innerClass = outerClass.new InnerClass();
        System.out.println(innerClass);
    }

    class OuterClass {
        // cannot be static for reasons
        class InnerClass {
            InnerClass() { }
        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.1-all.jar -c config.xml YOUR_FILE.java
Starting audit...
[ERROR] /var/tmp/Test.java:3:15: Unused local variable 'outerClass'. [UnusedLocalVariable]
Audit done.
Checkstyle ends with 1 errors.
```

---

I expect the variable `outerClass` to be correctly recognized as used.

This is not something we normally do in production code, but does exist in our unit tests.
